### ISSUE TITLE
[5.0] Prefetcher will block subsequent image loading from cache when prefetch thousands of urls

### ISSUE THREAD

**Issue by dreampiggy** [SDWebImage/SDWebImage#2630]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios
 Platform Version        | 12.1
 SDWebImage Version      | 5.0.0-beta5
 Integration Method      | cocoapods
 Xcode Version           | Xcode 10.1
 Repro rate              | all the time (100%)
 Repro with our demo prj | does it happen with our demo project? Yes with some code sample
 Demo project link       | see blow

### Issue Description and Steps

After upgrading to the 5.0.0-beta5, from the 4.4.6 version, I found a issue with the `SDWebImagePrefetcher` to prefetch image urls.

When using 5.0 prefetcher, to prefetch a large list of images urls, for example, 1000 urls. Then later normal image query from disk cache (using `sd_setImageWithURL:` from the imageView), will be blocking and delayed, until the previous 1000 image urls prefetch cache query finished.

Which means, if you write code to prefetch a list of urls, then using tableView or collectionView to load some of urls, the imageView will keeping blank for some period, until previous prefetch query cache logic finished. The disk load pressure is really high.

However, when using 4.4.6 version of SDWebImagePrefetcher, the subsequent imageView load will not been blocked, and prefetcher will start to load next image urls one by one, instead of query all of image urls from cache at the same time.

Sample code:
![11551605490_ pic_hd](https://user-images.githubusercontent.com/6919743/53693351-4584b280-3dda-11e9-90a3-01fc7a95cc38.jpg)

You can also modify some line of codes in the SDWebImage demo project.

+ First run without these code to make sure all images in the disk cache.
+ Next, add these code into `MasterViewController.viewDidLoad`, to prefetch 1000 urls before imageView load.
+ Run again, the already in-cached image will be blocked and imageView keep blank untill all 1000 urls prefetch finished. Which is not the designed behavior.

```objectivec
- (void)viewDidLoad {
    [super viewDidLoad];
    NSMutableArray<NSURL *> *imageURLs = [NSMutableArray arrayWithCapacity:1000];
    for (size_t i = 1; i <= 1000; i++) {
        NSString *url = [NSString stringWithFormat:@"http://via.placeholder.com/%zux%zu.jpg", i, i];
        [imageURLs addObject:[NSURL URLWithString:url]];
    }
    [[SDWebImagePrefetcher sharedImagePrefetcher] prefetchURLs:imageURLs];
}
```

@kinarob

**Comment by kinarob** [comment]
Yeah, I encountered this problem. When I used SDWebImage 4.4 in my project, because `SDWebImagePrefetcher` had a maximum number of limit in 4.x, it did not block threads, But only when our project was upgraded to 5.0, the first time we started our App and *Disk* had cached these pictures, there were many picture delayed to display  when scroll a lot of cells.

**Comment by kinarob** [comment]
Fix via [#2631](https://github.com/SDWebImage/SDWebImage/pull/2631). Thanks @dreampiggy
